### DISCUSSION TITLE
Use of RL algorithm for infrastructure comparison.

### DISCUSSION THREAD

**Discussion by AimanKhan1997** [zach401/acnportal#101] [Q&A]
Hi @sunash . I am working on the RL algorithm and it works fine for me. I was thinking of using this RL algorithm for infrastructure comparison as mentioned in [example 1.2](https://github.com/caltech-netlab/acnportal-experiments/blob/master/examples/1-Infrastructure-Evaluation/1.2-Comparing-Infrastructure-Designs/1.2-comparing-infrastructure-designs.ipynb). I was wondering how to call RL algorithm just like we called uncontrolled charging and least laxity factor algorithms. Is it possible, or should I change the environment in the RL code with the one provided in example 1.2
I want to check how the RL agent performs compared to those algorithms. Kindly let me know. Also, is it possible to get a plot of reward functions using plt.plot functions?

**Comment by sunash** [comment]
Hi @AimanKhan1997, thanks for posting. To run a trained RL agent, see the last cell of lessonx_training_running_rl_agent.ipynb. The key difference is that instead of using the _random_sim_builder helper function to construct evaluation_simulation, you will now want to construct the Simulator directly, as is done in the body of the run_experiment function in 1.2-comparing-infrastructure-designs.ipynb. This will allow you to initialize the Simulator with GMM-generated events and the same networks used in 1.2, while also allowing you to input your own GymTrainedAlgorithmVectorized instance. Just make sure you register an evaluation environment (that wraps the evaluation_simulation) to the evaluation algorithm, just like at the very end of lessonx.

The functionality to get reward functions via plt.plot isn't there yet, but you can calculate rewards over time directly from the run simulation. Also if you're comparing with 1.2, you'll want to use the functions in 1.2's run_experiment function (proportion_energy_delivered, num_swaps, etc).

Lastly, these only work for an already-trained RL agent, just to be clear.

Let me know if you have further questions or if anything needs clarifying.

**Reply by AimanKhan1997** [reply]
Hi @sunash , Sorry for the delay. I have been working on the simulator but still having issues implementing the run_experiment function. If possible, can please give clarify it a little more. Thanks again for the help.

**Reply by AimanKhan1997** [reply]
More specifically about how to use functions like proportion_energy_delivered, num_swaps etc

**Reply by sunash** [reply]
functions like proportion_energy_delivered are called on the Simulator after it has been run. It might help if you post your code here so I can see how you're currently constructing the run_experiment function for your use case.
